certbot/certbot/interfaces.py

```diff
@@ -70,35 +89,20 @@ class IPluginFactory(zope.interface.Interface):
 
     """
 
-    description = zope.interface.Attribute("Short plugin description")
-
-    def __call__(config, name):  # pylint: disable=signature-differs
-        """Create new `IPlugin`.
-
-        :param IConfig config: Configuration.
-        :param str name: Unique plugin name.
-
-        """
-
-    def inject_parser_options(parser, name):
-        """Inject argument parser options (flags).
+    description: str = NotImplemented
```

Comment:
Sadly, yes, it is not obvious way to define a contract on class parameter. We could go with a `@property`, but it will require more significant changes on the `disco` module to use the property at instance level.